File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if (nums.empty()) return 0;

        unordered_set<int> s;
        s.reserve(nums.size() * 2); // reduce rehashing
        for (int num : nums) {
            s.insert(num);
        }

        int longest = 0;
        for (int num : s) {
            if (!s.count(num - 1)) { // start of sequence
                int current = num;
                int count = 1;

                while (s.count(current + 1)) {
                    current++;
                    count++;
                }

                if (count > longest) longest = count;
            }
        }

        return longest;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
       Set<Integer> set = new HashSet<>();
       for(int x: nums) {
        set.add(x);
       }
       int longest = 0;
       int count = 0;
       for(int x: set) {
            if(!set.contains(x-1)) {
                count = 1;
                while(set.contains(x+count)) {
                        count++;
                }
            }
                longest = Math.max(count, longest);
       }

       return longest;
    }
}
